pixeltable/exprs/string_expr.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import Any, Optional, Union
+
+import sqlalchemy as sql
+
+import pixeltable.exceptions as excs
+import pixeltable.type_system as ts
+
+from .data_row import DataRow
+from .expr import Expr
+from .globals import StringOperator
+from .row_builder import RowBuilder
+from .sql_element_cache import SqlElementCache
+
+
+class StringExpr(Expr):
+    """
+    Allows expressions on strings
+    """
+
+    def __init__(self, operator: StringOperator, op1: Expr, op2: Expr):
+        super().__init__(ts.StringType(nullable=op1.col_type.nullable))
+        self.operator = operator
+        self.components = [op1, op2]
+        assert op1.col_type.is_string_type()
+        if operator in (StringOperator.ADD, StringOperator.MUL):
+            if operator == StringOperator.ADD and not op2.col_type.is_string_type():
+                raise excs.Error(
+                    f'{self}: {operator} on strings requires string type, but {op2} has type {op2.col_type}'
+                )
+            if operator == StringOperator.MUL and not op2.col_type.is_int_type():
+                raise excs.Error(f'{self}: {operator} on strings requires int type, but {op2} has type {op2.col_type}')
+        else:
+            raise excs.Error(
+                f'{self}: invalid operation {operator} on strings, only operators {StringOperator.ADD} and {StringOperator.MUL} are supported'
+            )
+        self.id = self._create_id()
+
+    @property
+    def _op1(self) -> Expr:
+        return self.components[0]
+
+    @property
+    def _op2(self) -> Expr:
+        return self.components[1]
+
+    def __repr__(self) -> str:
+        return f'{self._op1} {str(self.operator)} {str(self._op1)}'
+
+    def _equals(self, other: StringExpr) -> bool:
+        return self.operator == other.operator
+
+    def _id_attrs(self) -> list[tuple[str, Any]]:
+        return super()._id_attrs() + [('operator', self.operator.value)]
+
+    def sql_expr(self, sql_elements: SqlElementCache) -> Optional[sql.ColumnElement]:
+        left = sql_elements.get(self._op1)
+        right = sql_elements.get(self._op2)
+        if left is None or right is None:
+            return None
+        if self.operator == StringOperator.ADD:
+            return left.concat(right)
```

Comment:
Can't we use `REPEAT()` if it's a `StringOperator.MUL`?

Reply:
yes I need to figure out how to call postgres function directly will update PR in a bit